discussion_agents/cog/agent/react.py

```diff
@@ -219,3 +222,118 @@ def generate(self, observation_dict: Dict[str, str]) -> str:
             str: The generated response.
         """
         return self.agent.invoke(observation_dict)  # type: ignore
+
+
+class FEVERAgent(BaseAgent):
```

Comment:
Ideally, we want a single ReActAgent where we can pass in FEVER prompts or HotPotQA prompts or any other benchmark prompts.